backend/typescript/services/implementations/ReviewService.ts

```diff
@@ -0,0 +1,174 @@
+import { Sequelize } from "sequelize-typescript";
+import PgReview from "../../models/review.model";
+import PgBook from "../../models/book.model";
+import PgTag from "../../models/tag.model";
+import PgSeries from "../../models/series.model";
+import PgAuthor from "../../models/author.model";
+import PgPublisher from "../../models/publisher.model";
+import logger from "../../utilities/logger";
+import { sequelize } from "../../umzug";
+import testSequelize from "../../testUtils/testDb";
+import {
+  ReviewRequestDTO,
+  IReviewService,
+  ReviewResponseDTO,
+  Book,
+  Author,
+  Publisher,
+  Tag,
+} from "../interfaces/IReviewService";
+
+const Logger = logger(__filename);
+
+// Delete: can delete the book, book_author, and book_publisher
+// Get: return review - but it contains all other info
+
+class ReviewService implements IReviewService {
+  db: Sequelize;
+
+  constructor(isTest = false) {
+    if (isTest) {
+      this.db = testSequelize;
+    } else {
+      this.db = sequelize;
+    }
+  }
+
+  /* eslint-disable class-methods-use-this */
+  async createReview(review: ReviewRequestDTO): Promise<ReviewResponseDTO> {
+    let publisher: PgPublisher;
```

Comment:
Why do you keep publisher as a separate variable here? Why not handle it the same way as you did with authors

Reply:
I needed the publisher variable to add the relation at an outer scope